bundle/src/main/java/com/adobe/acs/commons/redirects/servlets/ImportRedirectMapServlet.java

```diff
@@ -0,0 +1,156 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.redirects.servlets;
+
+import com.adobe.acs.commons.redirects.models.RedirectRule;
+import org.apache.poi.ss.usermodel.*;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.request.RequestParameter;
+import org.apache.sling.api.resource.PersistenceException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.adobe.acs.commons.redirects.filter.RedirectFilter.*;
+import static com.adobe.granite.comments.AbstractCommentingProvider.JCR_CREATED_BY;
+import static org.apache.jackrabbit.JcrConstants.JCR_CREATED;
+import static org.apache.sling.api.resource.ResourceResolver.PROPERTY_RESOURCE_TYPE;
+
+/**
+ * A servlet to import redirect rules from an Excel spreadsheet
+ *
+ */
+@Component(service = Servlet.class, immediate = true, name = "ImportRedirectMapServlet", property = {
+        "sling.servlet.label=ACS AEM Commons - Import Redirects Servlet",
+        "sling.servlet.methods=POST",
+        "sling.servlet.selectors=import",
+        "sling.servlet.resourceTypes=" + ACS_REDIRECTS_RESOURCE_TYPE
+})
+public class ImportRedirectMapServlet extends SlingAllMethodsServlet {
+
+    private static final Logger log = LoggerFactory.getLogger(ImportRedirectMapServlet.class);
+    private static final long serialVersionUID = -3564475196678277711L;
+
+    @Override
+    protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response)
+            throws ServletException, IOException {
+
+        String path = request.getParameter("path");
+        Resource storageRoot = request.getResourceResolver().getResource(path);
+        log.debug("Updating redirect maps at {}", storageRoot.getPath());
+        Map<String, RedirectRule> jcrRules = getRules(storageRoot)
+                .stream().collect(Collectors.toMap(RedirectRule::getSource, r -> r,
+                        (oldValue, newValue) -> oldValue, LinkedHashMap::new)); // rules stored in crx
+        Map<String, RedirectRule> xlsRules = readEntries(getFile(request))
```

Comment:
try with resources on the returned input stream

Reply:
Changed